src/mysqlsh_helpers.py

```diff
@@ -35,84 +61,143 @@ class MySQL:
+        """Add a configuration file for mysqld and restart the mysql service."""
+        try:
+            # target file starts with 'z-' so it has priority over the default config file
+            shutil.copyfile("templates/mysqld.cnf", "/etc/mysql/mysql.conf.d/z-custom-mysqld.cnf")
+
+            restart_mysql_command = ["systemctl", "restart", "mysql"]
+            subprocess.check_output(restart_mysql_command, stderr=subprocess.PIPE)
+        except Exception as e:
+            logger.exception(
+                f"Failed to update mysql config for: {self.instance_address}", exc_info=e
+            )
+            raise MySQLUpdateConfigurationError(e.stderr)
 
     def configure_mysql_users(self):
         """Configure the MySQL users for the instance.
 
-        Creates base `root@%` and `clusteradmin@instance_address` user with the
+        Creates base `root@%` and `<server_config>@%` users with the
         appropriate privileges, and reconfigure `root@localhost` user password.
+
         Raises MySQLConfigureMySQLUsersError if the user creation fails.
         """
-        _script = (
+        # SYSTEM_USER and SUPER privileges to revoke from the root users
+        # Reference: https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#priv_super
+        privileges_to_revoke = (
+            "SYSTEM_USER",
+            "SYSTEM_VARIABLES_ADMIN",
+            "SUPER",
+            "REPLICATION_SLAVE_ADMIN",
+            "GROUP_REPLICATION_ADMIN",
+            "BINLOG_ADMIN",
+            "SET_USER_ID",
+            "ENCRYPTION_KEY_ADMIN",
+            "VERSION_TOKEN_ADMIN",
+            "CONNECTION_ADMIN",
+        )
+
+        commands = (
             "SET @@SESSION.SQL_LOG_BIN=0;",
-            f"CREATE USER '{self.cluster_admin_user}'@'{self.instance_address}' IDENTIFIED BY '{self.cluster_admin_password}';",
-            f"GRANT ALL ON *.* TO '{self.cluster_admin_user}'@'{self.instance_address}' WITH GRANT OPTION;",
             f"CREATE USER 'root'@'%' IDENTIFIED BY '{self.root_password}';",
             "GRANT ALL ON *.* TO 'root'@'%' WITH GRANT OPTION;",
+            f"CREATE USER '{self.server_config_user}'@'%' IDENTIFIED BY '{self.server_config_password}';",
+            f"GRANT ALL ON *.* TO '{self.server_config_user}'@'%' WITH GRANT OPTION;",
             "UPDATE mysql.user SET authentication_string=null WHERE User='root' and Host='localhost';",
             f"ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY '{self.root_password}';",
-            "REVOKE SYSTEM_USER ON *.* FROM root@'%';",
-            "REVOKE SYSTEM_USER ON *.* FROM root@localhost;",
+            f"REVOKE {', '.join(privileges_to_revoke)} ON *.* FROM root@'%';",
+            f"REVOKE {', '.join(privileges_to_revoke)} ON *.* FROM root@localhost;",
             "FLUSH PRIVILEGES;",
         )
 
         try:
-            self._run_mysqlcli_script(" ".join(_script))
+            logger.debug("Configuring MySQL users=")
```

Comment:
Is this `=` on purpose on the log message?

Reply:
removed in 37db8b57689305a169c643d92613f15416ef8710

Reply:
Nope, it was just a typo, but thanks for the link - I did not know you could do that